interop-testing/src/test/java/io/grpc/testing/integration/CompressionTest.java

```diff
@@ -0,0 +1,272 @@
+import io.grpc.ClientInterceptor;
+import io.grpc.Codec;
+import io.grpc.CompressorRegistry;
+import io.grpc.DecompressorRegistry;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerCall;
+import io.grpc.ServerCall.Listener;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.testing.TestUtils;
+import io.grpc.testing.integration.Messages.Payload;
+import io.grpc.testing.integration.Messages.PayloadType;
+import io.grpc.testing.integration.Messages.SimpleRequest;
+import io.grpc.testing.integration.Messages.SimpleResponse;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Tests that compression is turned on.  This leans on the fact that gzip is a standard compressor
+ * and expects it to be on.
+ */
+@RunWith(JUnit4.class)
+public class CompressionTest extends AbstractTransportTest {
+
+  private static int serverPort = TestUtils.pickUnusedPort();
+
+  private static DecompressorRegistry decompressors = DecompressorRegistry.newEmptyInstance();
+  private static CompressorRegistry compressors = CompressorRegistry.newEmptyInstance();
+
+  /** Start server. */
+  @BeforeClass
+  public static void startServer() {
+    decompressors.register(Fzip.INSTANCE, true);
+    compressors.register(Fzip.INSTANCE);
+    startStaticServer(
+        ServerBuilder.forPort(serverPort)
+            .compressorRegistry(compressors)
+            .decompressorRegistry(decompressors),
+        new ServerInterceptor() {
+          @Override
+          public <ReqT, RespT> Listener<ReqT> interceptCall(MethodDescriptor<ReqT, RespT> method,
+              ServerCall<RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+            Listener<ReqT> listener = next.startCall(method, call, headers);
+            // TODO(carl-mastrangelo): check that encoding was set.
+            call.setMessageCompression(true);
+            return listener;
+          }});
+  }
+
+  /** Stop server. */
+  @AfterClass
+  public static void stopServer() {
+    stopStaticServer();
+  }
+
+  @Before
+  public void beforeTests() {
+    Fzip.INSTANCE.readIn = 0;
+    Fzip.INSTANCE.readOut = 0;
+    Fzip.INSTANCE.writeOut = 0;
+    Fzip.INSTANCE.writeIn = 0;
+  }
+
+  @After
+  public void afterTests() {
+    // Just check that the compressor was involved.
+    assertTrue(Fzip.INSTANCE.anyRead || Fzip.INSTANCE.anyWritten);
+  }
+
+  @Test
+  public void compresses() {
```

Comment:
The test should cover all the possible values of the following parameters (32 cases in total):
```
boolean clientRegisteredCompressor,
boolean clientRegisteredDecompressor,
boolean serverRegisteredCompressor,
boolean serverRegisteredDecompressor,
boolean setMessageCompressor
```
and check whether compression is applied in both outgoing and incoming directions.